source/MaterialXRuntime/RtStage.cpp

```diff
@@ -174,7 +174,7 @@ RtPrim RtStage::createNodeDef(RtNodeGraph& nodeGraph,
     // Add an input per nodegraph input
     for (auto input : nodeGraph.getInputs())
     {
-        RtInput attr = nodedef.createInput(input.getName(), input.getType());
+        RtInput attr = nodedef.createInput(input.getName(), input.getType(), input.asA<RtInput>().isUniform() ? RtAttrFlag::UNIFORM : 0);
```

Comment:
publishes the uniform flag.

Reply:
A slightly more safe variation of this would be to use:

```
RtInput attr = nodedef.createInput(input.getName(), input.getType());
attr.setUniform(input.asA<RtInput>().isUniform());
```

This way you are not forcing the flags to be 0 if the input is not uniform, so if the default values for the flags changes to something other than 0 later it will still work here.